### PR TITLE
Added Demo for FisheyeGl Module

### PR DESCRIPTION
**Pull request by ccpandhare**
I have added a demo for the FisheyeGl module and it is now live at https://ccpandhare.github.io/image-sequencer/examples/fisheye

As #27 Suggests, I should add a 'crop+fisheyegl' demo. Should I add that in addition to this or remove this altogether and add just that?

### CODE REVIEW

**Comment by ccpandhare** [comment]
I have made the Demo like the one in https://jywarren.github.io/fisheyegl/example/
It can be seen at https://ccpandhare.github.io/image-sequencer/examples/fisheyegl

This has been made with the new UI System

**Comment by ccpandhare** [comment]
The link is https://ccpandhare.github.io/image-sequencer/examples/fisheye and not https://ccpandhare.github.io/image-sequencer/examples/fisheyegl, sorry for that.

**Comment by ccpandhare** [comment]
Once this is merged in, I will add a link to it in the description.
Also we should now have a gh-pages site for `publiclab/image-sequencer` too

**Comment by jywarren** [comment]
OK - this is super. I have some requests and thoughts that will be relevant to the UI system. 

First, specifically on this module's demo, there are some UI features (very recently added) which make the fisheye system much more useful:

* drag drop
* url hash used to store the settings, so you can pass a permalink that's a "preset" for a specific camera
* storing previous images in a grid below, to make batch processing easier

These are all in `main.js` here: https://github.com/jywarren/fisheyegl/blob/master/example/main.js

I think we could reformat that file to simply be directly re-usable somehow in this demo, so upgrades to that UI code can be available here too. But is this a sustainable workflow?

More generally, many modules may have very specific UIs designed just for their particular use, and which make them much more useful. How can we design the system to allow modules to supply a UI that's hyper-specific in a flexible but standard way?

Could `fisheye` also publish a separate `fisheye-ui`? or could there be a baked-in `module.ui`? 

What are your thoughts?

**Comment by ccpandhare** [comment]
I am not very sure how we can directly use the code from the module demo.

Maybe we can have the module specify UI components? How does that sound?

**Comment by jywarren** [comment]
Well, we could either release a second module for each, or include a UI
subcomponent in each.

**Comment by ccpandhare** [comment]
I have some concerns here, is Image Sequencer supposed to provide the UI?
Shouldn't the UI be implementation specific?

Maybe we could provide a standard UI - Users can choose to use it or not.

But honestly, I don't think Image Sequencer should be handling front end code (By that i mean the markup and the stylesheets).
It should definitely have an API for UI development, but maybe not provide UI itself, and serve it's main purpose of image processing.... What do you think about this aspect of Image Sequencer? What are your goals for Image Sequencer in this regard?

**Comment by jywarren** [comment]
Very good thoughts, thanks. 

I had to think about this for a bit. I think we definitely ought to provide a standard UI -- one which makes it very easy to try out and use the core modules.

But on consideration, what I'm proposing is not that we add all the features of `main.js` from `fisheyegl` to `image-sequencer`, but that we /remove/ them and come up with a standard way that `fisheyegl` -- or maybe `fisheyegl-ui`, a separate module that requires `fisheyegl` -- can provide a more complex HTML UI. Because I think many modules will have a very complex and refined way they enable people to interact with them in HTML. 

I think the UI that `fisheyegl` provides should have its own CSS + HTML if it wants, and that we provide a minimal standard CSS in `image-sequencer` so things don't look too inconsistent. 

How does that sound? Do you think modules should provide their own UI, or should have a corresponding `_____-ui` module that provide the UI? I lean towards providing their own as a default; they could opt for a separate UI module if they like, which then wraps the original module.

**Comment by jywarren** [comment]
If you like this proposal, `fisheyegl`'s UI could be tidied up and modularized from its `main.js` file, and then we could test out this arrangement using this as the first module to provide its own UI. What do you think?

File: test/image-manip.js [hunk 1]
```diff
@@ -10,8 +10,8 @@ require('../src/ImageSequencer.js');
 //require image files as DataURLs so they can be tested alike on browser and Node.
 var sequencer = ImageSequencer({ ui: false });
 
-var test_png = require('../examples/test.png.js');
-var test_gif = require('../examples/test.gif.js');
+var test_png = require('../examples/images/test.png.js');
```
Comment (jywarren) [inline test/image-manip.js]: Perhaps we should keep test images in `/test/images/` even if it's redundant... what do you think?

File: test/image-sequencer.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ test('loadImages loads a DataURL image and creates a step.', function (t){
 
 if(!sequencer.options.inBrowser)
   test('loadImage loads an image from URL and creates a step. (NodeJS)', function (t){
-    sequencer.loadImage('URL','https://ccpandhare.github.io/image-sequencer/examples/red.jpg', function(){
+    sequencer.loadImage('URL','https://ccpandhare.github.io/image-sequencer/examples/images/red.jpg', function(){
```
Comment (jywarren) [inline test/image-sequencer.js]: I like that you want to test this, but I'm concerned about making tests rely on an internet connection. Is there any other way to do this? I guess we can leave it as is for now, but you know what I mean? It just ... feels a bit wrong i guess... but I can't come up with a really really good reason why :-)

**Comment by jywarren** [comment]
I guess also, what does this mean for this PR? Could we, or should we:

1. just merge this in and think about the custom UI provision separately
2. use this as the first instance of a module providing its own UI -- test out this idea?

Thanks for thinking through this carefully. It really does get us deeply into the UI phase of the work :-)

File: test/image-manip.js (hunk 1 above)
Comment (ccpandhare) [inline test/image-manip.js]: Yes that does sound good. I shall do that.

File: test/image-sequencer.js (hunk 2 above)
Comment (ccpandhare) [inline test/image-sequencer.js]: Yes, I had thought about this, too.
But I just couldn't come up with a substitute for this.
Maybe we could run this test if and only internet access is available?

**Comment by jywarren** [comment]
Sure! But let's output a note that we're skipping due to no internet.

**Comment by ccpandhare** [comment]
Thanks for the review! To sum up and suggest further changes:
- We had a discussion about whether image sequencer should provide a UI or not. We concluded that Image Sequencer should provide a basic UI and that modules should have the liberty to suggest UI.
    *Further thoughts on this:* I think the module should specify only the elements and the general CSS of image-sequencer should be responsible for styling. Which means modules specify the UI but not style it. What do you think?

**Comment by ccpandhare** [comment]
- I think this is a major upgrade and should go in a different PR

**Comment by jywarren** [comment]
Agreed on everything except that sometimes CSS will be needed for
functional reasons of UI like positioning. So the CSS convention should be
a recommendation only? Or it's for font styling and such, but not all CSS.

Great! So let's merge this and move our recap to a new issue + PR. Thanks!!

**Comment by ccpandhare** [comment]
Okay!

File: test/image-manip.js (hunk 1 above)
Comment (ccpandhare) [inline test/image-manip.js]: Fixed this in an upcoming commit

File: test/image-sequencer.js (hunk 2 above)
Comment (ccpandhare) [inline test/image-sequencer.js]: I am checking for internet access in an upcoming commit.
Hope that helps.

**Comment by ccpandhare** [comment]
If you are fine with the changes I made, I shall merge this in!

**Pull request merged**